Index: tool/pylib/misc/cldr.py
===================================================================
--- tool/pylib/misc/cldr.py	(revision 27577)
+++ tool/pylib/misc/cldr.py	(working copy)
@@ -151,7 +151,16 @@
 
     return data
 
+def extractOrientation(tree):
+    data = {}
 
+    layoutOrientationCharactersNode = tree.find("layout/orientation")
+    if layoutOrientationCharactersNode != None:
+        data['cldr_layout_orientation_characters'] = layoutOrientationCharactersNode.attrib["characters"]
+
+    return data
+
+
 def parseCldrFile(filename, outputDirectory=None):
     tree = ElementTree.parse(filename)
 
@@ -172,6 +181,7 @@
 
     data.update(extractDelimiter(tree))
     data.update(extractNumber(tree))
+    data.update(extractOrientation(tree))
 
     return data
     
